HEN_HOUSE/egs++/egs_interpolator.cpp

```diff
@@ -99,6 +99,13 @@ void EGS_Interpolator::initialize(int nbin, EGS_Float Xmin, EGS_Float Xmax,
         b[j] = (values[j+1]-values[j])*bx;
         a[j] = values[j] - b[j]*(xmin + dx*j);
     }
+    /**************************************************** 
+       Extra subinterval at top of interval, taking care 
+       of round-off errors via extrapolation.
+       Mimics PEGS4 PWLF QFIT approach.
+    *****************************************************/
+    a[n] = a[n-1];
+    b[n] = b[n-1];
```

Comment:
Does this change have any implications for other areas of the code that use interpolators?

Reply:
I don't think so, because the allocation is extended by one bin, and the last element duplicated in the new bin. Existing code would loop until bin `n` and not see the extra bin appended here.